### ISSUE TITLE
Operations with >13 qubits crash julia session

### ISSUE THREAD

**Issue by mmikhasenko** [QuantumBFS/Yao.jl#256]
Trying to run an example of the ground state of the Heisenberg hamiltonian for the [tutorial](https://github.com/QuantumBFS/tutorials/blob/master/src/index.md)
```julia
function grounde(nbit)
    hami = heisenberg(nbit)
    te = time_evolve(hami |> cache, -10im)
    reg = rand_state(nbit)
    energy(reg) = real(expect(hami, reg))
    reg |> te |> normalize!
    energy(reg)    
end

grounde(2) # works great
grounde(13) # works
grounde(14) # crashes julia session

# it is not a simple problem of the memory limit since
rand(1<<14, 1<<14) # works
```
The problem happens at `reg |> te |> normalize!` line.
`KrylovKit.eigsolve` applied to `mat(hami)` would also crash everything.

```
Windows 10
Julia 1.5
[5872b779] Yao v0.6.3
[7a06699c] YaoExtensions v0.2.2
```

**Comment by GiggleLiu** [comment]
This looks tricky, it looks like a platform related issue because the above code works on my laptop.
```julia
julia> using Yao

julia> function grounde(nbit)
           hami = heisenberg(nbit)
           te = time_evolve(hami |> cache, -10im)
           reg = rand_state(nbit)
           energy(reg) = real(expect(hami, reg))
           reg |> te |> normalize!
           energy(reg)    
       end
grounde (generic function with 1 method)

julia> grounde(2) # works great
-6.0

julia> grounde(13) # works
-22.518337318976034

julia> grounde(14) # crashes julia session
-25.05419811594903
```

My platform is Ubuntu 20.04, Julia 1.5
```
(@v1.5) pkg> st Yao YaoExtensions
Status `~/.julia/environments/v1.5/Project.toml`
  [5872b779] Yao v0.6.3 `~/.julia/dev/Yao`
  [7a06699c] YaoExtensions v0.2.2 `~/.julia/dev/YaoExtensions`
```

 Could you please check if it is the `mat(hami)` that breaks the Julia REPL?

**Comment by mmikhasenko** [comment]
Hi, thanks for the check. `mat(hami)` works, it gives elements of the sparse matrix.

**Comment by GiggleLiu** [comment]
Ok. If it is not the problem of matrix construction. It could be the issue of using the following two packages.
* The krylov solver is from: https://github.com/Jutho/KrylovKit.jl
* The expv function for computing time evolution is from https://github.com/SciML/ExponentialUtilities.jl

If I were you, I will **copy and shuffle the indices** of the generated sparse matrix.
```julia
sp2 = SparseMatrixCSC(SparseMatrixCSC(sp')')
```
and then use `sp2` to check if the `eigsolve` and the `expv` in the two packages are working properly.

**Comment by Roger-luo** [comment]
Does the time evolution work for you? When you run `|> te`

**Comment by Roger-luo** [comment]
I cannot reproduce this either, but I think both me and @GiggleLiu are using Linux/Mac OS, thus, it could be some bug on Windows specific. And is there an error msg? Is it segment fault?

**Comment by mmikhasenko** [comment]
> * The krylov solver is from: https://github.com/Jutho/KrylovKit.jl

Yes, likely that, since `eigensolve(sp, 1, :SR)` gives the problem.
It would be nice to construct a minimal example without using Yao what would crash the same way
How can get a random sparse matrix of the same dimension? Should it crash then... (?)

**Comment by mmikhasenko** [comment]
> If I were you, I will **copy and shuffle the indices** of the generated sparse matrix by
> 
> ```julia
> sp2 = SparseMatrixCSC(SparseMatrixCSC(sp')')
> ```

Just tried that, still crashes.
I cannot see the error message since the session and the terminal is killed immediately after the crash.

**Comment by mmikhasenko** [comment]
I just caught the crashing screen.
```julia
using Yao
using YaoExtensions
using KrylovKit
using SparseArrays
# 
hami = heisenberg(14)
m = mat(hami)
sp2 = SparseMatrixCSC(SparseMatrixCSC(m')')

function failing_function(sp)
    size(sp)
    length(sp)
    eigsolve(sp2, 1, :SR)
end
```
![image](https://user-images.githubusercontent.com/22725744/91709615-4808c980-eb83-11ea-8285-ca6224353425.png)

Something with `libopenblas64_.DLL`...

**Comment by Roger-luo** [comment]
I think this is likely a bug in Julia and is windows specific, I can reproduce this on v1.5.1 on windows now. I'll see if I can fix it or report it to Julia repo if I can't shortly. Thanks for reporting.

**Comment by Roger-luo** [comment]
This should be fine on v1.4, if you need to calculate something, please use Julia v1.4 for now. @mmikhasenko

**Comment by mmikhasenko** [comment]
good to know, thanks for the quick investigation.

**Comment by Roger-luo** [comment]
this should be fixed on 1.7 now
